### DISCUSSION TITLE
Working with Microsoft/Apple Japanese IME

### DISCUSSION THREAD

**Discussion by nogurenn** [manna-harbour/miryoku#331] [Q&A] [answered]
Beginner miryoku user here. I've been trying to use the IME input for language learning. Is there an additional setup that works for you? Like maybe remapping some OS shortcuts? Something about having to press Shift when pressing Caps lock on the Miryoku. The IME changes input mode correctly using the caps lock shortcuts from my regular Logitech keyboard.

Windows 11 ([official reference, English 101/102 layout](https://support.microsoft.com/en-us/windows/microsoft-japanese-ime-da40471d-6b91-4042-ae8b-713a96476916#ID0EBBL=English_keyboard_(101/102_key)))
~~1. `` Alt + Backquote (`) ``  ✅ [Cycle to next OS input language] 
2. `Ctrl + Caps lock` needs Shift but works [Switch to Hiragana]
3. `Alt + Caps lock` needs Shift but works [Switch to Katakana]
4. `Shift + Caps lock` ❌ [Eisu equivalent]
    * Cannot activate half-width alphanumeric, so I'm stuck with hiragana/katakana until I cycle to another input language and back to Japanese IME~~
See updated tests in comment.

KB: choc fifi (pashutk)
MC: pro micros

**Comment by nogurenn** [comment]
Restored all relevant shortcuts and tested again. I think I just don't understand the resulting key inputs of Miryoku's CAPS and SHIFT+CAPS. Would someone kindly enlighten me? Am I missing something below? 🙏🏼 

1. `GUI + Space` - cycle OS input
2. `` Alt + Backquote (`) `` - toggle between last used and Half-width Alphanumeric / Direct Input
    * ! ohayougozaimasu.
    * ！　おはようございます。
3. `Caps lock` - toggle between caps lock and lowercase, does **not** activate Windows caps lock overlay
4. `Ctrl + Caps lock` - activate caps lock but cannot deactivate back using the same, does **not** activate Windows overlay
    * ! OHAYOUGOZAIMASU.
5. `Alt + Caps lock` - same as (4)
    * ! OHAYOUGOZAIMASU.
6. `Alt + Ctrl + Caps lock` - same as (4)
    * ! OHAYOUGOZAIMASU.
7. `Shift + Caps lock` - toggle between caps lock and lowercase, activates Windows overlay
    * ! OHAYOUGOZAIMASU.
    * ! ohayougozaimasu.
8. `Shift + Ctrl + Caps lock` - switch to Hiragana
    * ！　おはようございます。
9. `Shift + Alt + Caps lock` -  switch to Katakana
    * ！　オハヨウゴザイマス。
10. `Shift + Alt + Ctrl + Caps lock` - same as (7)
    * ! OHAYOUGOZAIMASU.
    * ! ohayougozaimasu.

**Comment by manna-harbour** [comment, marked as answer]
See https://github.com/manna-harbour/miryoku_qmk/tree/miryoku/users/manna-harbour_miryoku#caps-word and https://github.com/manna-harbour/miryoku_zmk#caps-word. The key labelled Caps Lock in the diagram https://github.com/manna-harbour/miryoku/tree/master/docs/reference#nav is Caps Word, and when combined with Shift is Caps Lock. So it's not possible to produce Shift + Caps Lock.

If you need Caps Lock instead of Caps Word, you can substitute a custom Nav layer with https://github.com/manna-harbour/miryoku/discussions/85 and replace Caps Word with Caps Lock. Alternatively, change the key bindings to not be based on Caps Lock.

**Reply by nogurenn** [reply]
I will check the differences of Caps Word from Caps Lock. Thank you!
